test/mpi/main.cc

```diff
@@ -391,8 +391,8 @@ int main(int argc, char *argv[])
     int rank;
     int failed = 0;
 
-    UccTestMpi test(argc, argv, UCC_THREAD_SINGLE, teams, cls);
     ProcessArgs(argc, argv);
+    UccTestMpi test(argc, argv, UCC_THREAD_SINGLE, teams, cls);
```

Comment:
In that order mtype_str_to_type will not work for cuda buffers, because it uses ucc_mc_available function

Reply:
fixed by creating teams explicitly after parsing args